aliases/available/general.aliases.bash

```diff
@@ -2,19 +2,55 @@ cite about-alias
 about-alias 'general aliases'
 
 # List directory contents
-alias sl=ls
-alias ls='ls -G'        # Compact view, show colors
-alias la='ls -AF'       # Compact view, show hidden
-alias ll='ls -al'
-alias l='ls -a'
-alias l1='ls -1'
+# Alias source: http://tldp.org/LDP/abs/html/sample-bashrc.html
+
+local LS_COLOR_OPTION
+
+case "$OSTYPE" in
+  linux*)
+    # Set Linux color option
+    LS_COLOR_OPTION="--color=auto"
+    ;;
+  darwin*)
+    # Set BSD color optio
+    LS_COLOR_OPTION="-G"
+    #Check if coreutils version of if exists
+    if [[ -x "/usr/local/opt/coreutils/libexec/gnubin/ls" ]]; then
+      # Check if coreutils path is in $PATH
+      if [[ ":$PATH:" == *":/usr/local/opt/coreutils/libexec/gnubin:"* ]]; then
+        # Set Linux color option
+        LS_COLOR_OPTION="--color=auto"
+      fi
+    fi
+    ;;
+  *)
+    # Use Linux color option as fallback
+    LS_COLOR_OPTION="--color=auto"
+    ;;
+esac
+
+
+# # Add colors for filetype and  human-readable sizes by default on 'ls':
+alias l="ls -a $LS_COLOR_OPTION"            # Standard
+alias lx="ls -lXB $LS_COLOR_OPTION"         # Sort by extension.
+alias lk="ls -lSr $LS_COLOR_OPTION"         # Sort by size, biggest last.
+alias lt="ls -ltr $LS_COLOR_OPTION"         # Sort by date, most recent last.
+alias lc="ls -ltcr $LS_COLOR_OPTION"        # Sort by/show change time,most recent last.
+alias lu="ls -ltur $LS_COLOR_OPTION"        # Sort by/show access time,most recent last.
+
+# # The ubiquitous 'll': directories first, with alphanumeric sorting:
+alias ll="ls -lv --group-directories-first $LS_COLOR_OPTION"
```

Comment:
This does not work with the stock OS X version either, the `--group-directories-first` option is not supported.